File: 2015/day01/perl/day01p2.pl
#2025 Levi D. Smith

open(f, "../input01.txt");
while($strLine = <f>) {
#  print "$strLine\n";
  chomp($strLine);
  @chars = split(//, $strLine);
  $iValue = 0;
  $i = 0;
  foreach my $c (@chars) {
    if ($c eq '(') {
        $iValue += 1;
    } elsif ($c eq ')') {
        $iValue -= 1;
    }
    
    if ($iValue == -1) {
      print ($i + 1);
      print "\n";
      last;
    }
    $i += 1;
  }
}

File: 2025/day01/ruby/day01p2.rb
#2025 Levi D. Smith <developer@levidsmith.com>
f = open("../input01.txt")

iCurrent = 50
iPassword = 0

MIN_VALUE = 0
MAX_VALUE = 99 
TARGET_VALUE = 0

while(strLine = f.gets)
#  puts strLine
  strLine.chomp!
  strDir = strLine[0]
  iValue = strLine[1..strLine.size].to_i

  case strDir
  when 'L'
    while (iValue > 0)
      iCurrent -= 1 
      if (iCurrent == TARGET_VALUE) 
        iPassword += 1
      elsif (iCurrent == MIN_VALUE - 1)
        iCurrent = MAX_VALUE 
      end
      iValue -= 1
    end
  when 'R'
    while (iValue > 0)
      iCurrent += 1 
      if (iCurrent == MAX_VALUE + 1) 
        iCurrent = MIN_VALUE 
      end
      if (iCurrent == TARGET_VALUE) 
        iPassword += 1
      end
      iValue -= 1
    end
  end



#  puts "password #{iPassword}"
#  puts "current #{iCurrent}"

end
f.close

puts iPassword
